File: 0609. Find Duplicate File in System/solution.py
class Solution:
    def findDuplicate(self, paths: List[str]) -> List[List[str]]:
        duplicates = collections.defaultdict(list)
        result = []
        
        for path in paths:
            tokens = path.split(' ')
            base_path, files = tokens[0], tokens[1:]
            
            for file in files:
                filename, contents = file[:-1].split('(')
                duplicates[contents].append(base_path + '/' + filename)
                
        for content in duplicates:
            if len(duplicates[content]) > 1:
                result.append(duplicates[content])
        
        return result
                

File: 0609. Find Duplicate File in System/solution.cpp
class Solution {
public:
    vector<vector<string>> findDuplicate(vector<string>& paths) {
        vector<vector<string>> result;
        unordered_map<string, vector<string>> duplicates;
        
        for (auto path: paths) {
            stringstream ss(path);
            string token, basePath, fileName, fileContents;
            
            ss >> basePath;
            
            while (ss >> token) {
                size_t contentsStart = token.find('(');
                fileName = token.substr(0, contentsStart);
                fileContents = token.substr(contentsStart);
                duplicates[fileContents].push_back(basePath + '/' + fileName);
            }
            
        }
        
        for (auto duplicate : duplicates) {
            if (duplicate.second.size() > 1)
                result.push_back(duplicate.second);
        }

        return result;
    }
};
